CommandDotNet/MicrosoftCommandLineUtils/CommandLineApplication.cs

```diff
@@ -395,15 +378,30 @@ public string GetFullNameAndVersion()
 
         public void ShowRootCommandFullNameAndVersion()
         {
-            var rootCmd = this;
-            while (rootCmd.Parent != null)
-            {
-                rootCmd = rootCmd.Parent;
-            }
-
+            var rootCmd = GetParentCommands().Last();
             Out.WriteLine(rootCmd.GetFullNameAndVersion());
             Out.WriteLine();
         }
+        
+        private IEnumerable<CommandLineApplication> GetSelfAndParentCommands()
+        {
+            for (CommandLineApplication c = this; c != null; c = c.Parent)
```

Comment:
I liked this approach for looping through a hierarchy.  I think I'll use it from now on.  I'm not sure it's as immediately intuitive as a while statement but it's terser and the intent is obvious from the method it's in.

Reply:
Nice. I find this more easy to understand